### ISSUE TITLE
Handle InvalidAuthTokenError with logout/reauth

### ISSUE THREAD

**Issue by shyba** [pixelated/pixelated-user-agent#485]
After a while logged in, auth token expires. This exception can trigger a logout or reauth.
```
2015-10-05 11:24:35 [leap.soledad.client.api] ERROR Soledad exception when syncing!
*--- Failure #39794 ---
Failure: leap.soledad.common.errors.InvalidAuthTokenError: 
*--- End of Failure #39794 ---
```

Definition of done
==========
After X hours (the time of token expiration) the account is able to recover from this error. Either with automatic logout or re-auth. In a nutshell: no user interaction to fix it.

**Comment by bwagnerr** [comment]
Hey, this is what happened when I would be logged out @shyba @fbernitt:

Traceback (most recent call last):
  File "/home/vagrant/user-agent-venv/bin/pixelated-user-agent", line 9, in <module>
    load_entry_point('pixelated-user-agent==0.1', 'console_scripts', 'pixelated-user-agent')()
  File "/vagrant/service/pixelated/application.py", line 108, in initialize
    deferred.addErrback(_quit_on_error)
  File "/home/vagrant/user-agent-venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 328, in addErrback
    errbackKeywords=kw)
  File "/home/vagrant/user-agent-venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 306, in addCallbacks
    self._runCallbacks()
--- <exception caught here> ---
  File "/home/vagrant/user-agent-venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/vagrant/service/pixelated/application.py", line 101, in _quit_on_error
    reactor.stop()
  File "/home/vagrant/user-agent-venv/local/lib/python2.7/site-packages/twisted/internet/base.py", line 581, in stop
    "Can't stop reactor that isn't running.")
twisted.internet.error.ReactorNotRunning: Can't stop reactor that isn't running.
2015-10-13 13:52:10 [twisted] CRITICAL

**Comment by bwagnerr** [comment]
But yes, it was dead :)

**Comment by bwagnerr** [comment]
Just happened again, InvalidAuthToken killed the service as we wanted, I would say this is working :)

2015-10-13 15:09:40 [leap.soledad.client.api] ERROR Soledad exception when syncing!
*--- Failure #153 ---
Failure: leap.soledad.common.errors.InvalidAuthTokenError:
*--- End of Failure #153 ---

(user-agent-venv)vagrant@leap-wheezy:/vagrant/service$

**Comment by shyba** [comment]
Hey @bwagnerr 
Did you check how the UI behaves with dispatcher poking a dead container?
I tried to check that on Friday, but we got that environment error.

My guess is that api calls will return 302 and the webui javascript won't redirect to logout, but I can be wrong.
This is what happens today on wazokazi if we stop a container (by shell) and keep the browser's tab open:
![screenshot from 2015-10-13 17 40 08](https://cloud.githubusercontent.com/assets/1302078/10467697/9f57defa-71d1-11e5-8c41-be0ca1f5de95.png)
Everything I click returns an error and I only get to logout screen when I refresh.
Maybe the UI should consider 302 as a full refresh?
Thoughts?
